File: 27. Remove Element/27. Remove Element.py
from typing import List

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        filter_len = len(filter_nums := list(filter(lambda x: x != val, nums)))
        nums[0: filter_len] = filter_nums
        return filter_len
        
if __name__ == '__main__':
    c = Solution()
    nums = [3,2,2,3]
    print(c.removeElement(nums, 3))
    print(nums)

File: 27.RemoveElement/27.RemoveElement.go
package main

import "fmt"

func removeElement(nums []int, val int) int {
	i := 0
	for j, v := range nums {
		if nums[j] != val {
			nums[i] = v
			i++
		}
	}
	return i
}

func main() {
	test := []int{3, 2, 2, 3}
	fmt.Println(removeElement(test, 3))
}
